src/containers/TemplateBuilder/Templates.tsx

```diff
@@ -90,36 +105,83 @@ const ViewEditButton: React.FC<CellProps> = ({ template: { id } }) => {
+        }
+      )
+      const data = await res.blob()
+      var a = document.createElement('a')
+      a.href = window.URL.createObjectURL(data)
+      a.download = `${snapshotName}.zip`
+      a.click()
+      // Delete the snapshot cos we don't want snapshots page cluttered
+      // with individual templates
+      await fetch(getServerUrl('snapshot', { action: 'delete', name: snapshotName }), {
+        method: 'POST',
+        headers: { Authorization: `Bearer ${JWT}` },
+      })
+      showToast()
+    } else showToast({ style: 'error', title: 'Problem exporting template' })
+  }
 
   return (
     <div key="export">
+      <Confirm
+        open={open}
+        // Prevent click in Input from closing modal
+        onClick={(e: any) => e.stopPropagation()}
+        content={
+          <div style={{ padding: 10, gap: 10 }} className="flex-column">
+            <p>Commit and export template?</p>
+            <p>
+              By exporting this template now, you will be committing the current version. To make
+              any further changes, you will need to duplicate it and start a new template version.
+            </p>
+            <div className="flex-row-start-center" style={{ gap: 10 }}>
+              <label>Please provide a commit message:</label>
+              <Input
+                value={commitMessage}
+                onChange={(e) => setCommitMessage(e.target.value)}
+                style={{ width: '60%' }}
+              />
+            </div>
+          </div>
+        }
+        onCancel={() => setOpen(false)}
+        onConfirm={async () => {
+          const versionId = getTemplateVersionId()
+          await updateTemplate(template as any, {
+            versionId,
+            versionComment: commitMessage,
+            versionTimestamp: DateTime.now().toISO(),
+          })
+          setOpen(false)
+          await doExport(versionId)
+        }}
+      />
       <div
         className="clickable"
         onClick={async (e) => {
           e.stopPropagation()
-          if (await exportTemplate({ id, snapshotName })) {
-            const res = await fetch(
-              getServerUrl('snapshot', { action: 'download', name: snapshotName }),
-              {
-                headers: { Authorization: `Bearer ${JWT}` },
-              }
-            )
-            const data = await res.blob()
-            var a = document.createElement('a')
-            a.href = window.URL.createObjectURL(data)
-            a.download = `${snapshotName}.zip`
-            a.click()
-            // Delete the snapshot cos we don't want snapshots page cluttered
-            // with individual templates
-            await fetch(getServerUrl('snapshot', { action: 'delete', name: snapshotName }), {
-              method: 'POST',
-              headers: { Authorization: `Bearer ${JWT}` },
-            })
-          }
+          if (isTemplateUnlocked(template)) setOpen(true)
+          else doExport()
```

Comment:
We only need to prompt for uncommitted versions -- already committed ones just get exported straight out like they do currently.